src/api/middleware/controllers/user.js

```diff
@@ -0,0 +1,65 @@
+const User = require("../../../models/User");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+//Signup
+const signup = async (req, res) => {
+  try {
+    const { username, firstname, lastname, email, password } = req.body;
+    const data = {
+      username,
+      email,
+      firstname,
+      lastname,
+      password: await bcrypt.hash(password, 10),
+    };
+    const user = await User.create(data);
+    return res
+      .status(201)
+      .json({ message: "User created successfully", user: user });
+  } catch (error) {
+    return res.status(509).json({
+      message: "An error occured while processing your request",
+    });
+  }
+};
+
+//login
+const login = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    const user = await User.findOne({ where: { email: email } });
+
```

Comment:
Same thing here, you do not need `where` object pattern of filtering with Mongoose, it is only required when working with Sequelize.

So this should be `const user = await User.findOne({email: email });`.
See https://mongoosejs.com/docs/api.html#model_Model-findOne 

Please check other instances you are using the `where` when filtering and remove it.